exporter/loadbalancingexporter/log_exporter.go

```diff
@@ -76,25 +76,45 @@ func (e *logExporterImp) Shutdown(context.Context) error {
 
 func (e *logExporterImp) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
 	var errs error
+	endpointToLogData := make(map[string]plog.Logs)
 	batches := batchpersignal.SplitLogs(ld)
-	for _, batch := range batches {
-		errs = multierr.Append(errs, e.consumeLog(ctx, batch))
+
+	// Map the logs to their respective endpoints.
+	for batch := range batches {
+		// Each batch contains at most one trace ID.
+		traceID := traceIDFromLogs(ld)
+		balancingKey := traceID
+		if traceID == pcommon.NewTraceIDEmpty() {
+			// every log may not contain a traceID
+			// generate a random traceID as balancingKey
+			// so the log can be routed to a random backend
+			balancingKey = random()
+		}
+
+		endpoint := e.loadBalancer.Endpoint(balancingKey[:])
+		if _, ok := endpointToLogData[endpoint]; ok {
+			// append
+			for i := 0; i < batches[batch].ResourceLogs().Len(); i++ {
+				batches[batch].ResourceLogs().At(i).CopyTo(endpointToLogData[endpoint].ResourceLogs().AppendEmpty())
+			}
+		} else {
+			newLog := plog.NewLogs()
+			for i := 0; i < batches[batch].ResourceLogs().Len(); i++ {
+				batches[batch].ResourceLogs().At(i).CopyTo(newLog.ResourceLogs().AppendEmpty())
+			}
```

Comment:
Similar question about copying here. For traces I see some fanout logic, so I believe it is appropriate to copy for traces